spec/policies/run_policy_spec.rb

```diff
@@ -0,0 +1,90 @@
+describe RunPolicy do
+  let(:run) { create(:run) }
+  
+  let(:problem_true) { create(:problem, runs_visible: true) }
+  let(:problem_false) { create(:problem, runs_visible: false) }
+
+  let(:false_checkbox_run) { create(:run, problem: problem_false) }
+  let(:true_checkbox_run) { create(:run, problem: problem_true) }
```

Comment:
I am not sure the names are really clear here. I would use something like `problem_with_visible_runs` and `problem_without_visible_runs` and `public_run` and `private_run`